### PR TITLE
[Tracer][Logs] ILogger sample in Azure Functions

### PR DESCRIPTION
**Pull request by pierotibou**
## Summary of changes
Added a sample for agentless logging in Azure Functions as I was investigating an issue

## Reason for change
Showcase an example of setup

## Implementation details
Heavily inspired from Steven's work.

### CODE REVIEW

File: tracer/samples/AzureFunctionsWithAgentLessLogging/FunctionWithLogging/run.ps1 [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    Set-EnvironmentVariableForProcess -name "CORECLR_ENABLE_PROFILING" -value "1"
+    Set-EnvironmentVariableForProcess -name "CORECLR_PROFILER" -value "{846F5F1C-F9AE-4B07-969E-05C26BC060D8}"
+    Set-EnvironmentVariableForProcess -name "CORECLR_PROFILER_PATH" -value $ClrProfilerPath
+
+    Set-EnvironmentVariableForProcess -name "DD_DOTNET_TRACER_HOME" -value $DotNetTracerHome
+
+    BuildAndRunSample
+}
+
+if ($IsMacOs) {
```
Comment (bouwkast) [inline tracer/samples/AzureFunctionsWithAgentLessLogging/FunctionWithLogging/run.ps1:120]: Nice, I must've missed this as I didn't know it was provided along with `IsWindows`/`IsLinux`

File: tracer/samples/AzureFunctionsWithAgentLessLogging/FunctionWithLogging/run.ps1 [hunk 2]
```diff
@@ -0,0 +1,133 @@
+<#
+.SYNOPSIS
+    A script that allows the user to run C# sample applications to demonstrate
+    the setup and usage of Datadog's .NET Tracer log collection and log trace ID injection in a function
+.PARAMETER AppDirectory
+    The application directory - defaults to the location of this script and need not be set
+    by the user unless samples are moved around.
+.PARAMETER ApiKey
+    The Datadog API key (only required when -Agentless is set)
```
Comment (bouwkast) [inline tracer/samples/AzureFunctionsWithAgentLessLogging/FunctionWithLogging/run.ps1:9]: nitpick but the `(only required when -Agentless is set)` can be removed for this

```suggestion
    The Datadog API key
```

File: tracer/samples/AzureFunctionsWithAgentLessLogging/FunctionWithLogging/run.ps1 [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    pwsh BuildAndRunSample.ps1 -ApiKey YOUR_API_KEY_HERE
+#>
+
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory = $false)]
+    [System.String]
+    $AppDirectory, # only necessary if the script is removed from its current directory layout
+
+    [Parameter(Mandatory = $false)]
```
Comment (bouwkast) [inline tracer/samples/AzureFunctionsWithAgentLessLogging/FunctionWithLogging/run.ps1:20]: This one might actually be mandatory? Maybe not though if they have it set outside?

**Review by bouwkast — APPROVED** [review]
Nice, just a couple of comments on the comments in the script

File: tracer/samples/AzureFunctionsWithAgentLessLogging/FunctionWithLogging/run.ps1 [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+## Configure logs injection by setting up Unified Service Tagging
+## cf https://docs.datadoghq.com/tracing/other_telemetry/connect_logs_and_traces/dotnet/?tab=serilog
+Set-EnvironmentVariableForProcess -name "DD_LOGS_INJECTION" -value "true"
+Set-EnvironmentVariableForProcess -name "DD_ENV" -value "dev"
+Set-EnvironmentVariableForProcess -name "DD_SERVICE" -value "LogsInjectionInFunctionSamples"
+Set-EnvironmentVariableForProcess -name "DD_VERSION" -value "1.0.0"
+Set-EnvironmentVariableForProcess -name "DD_API_KEY" -value $ApiKey
+Set-EnvironmentVariableForProcess -name "DD_LOGS_DIRECT_SUBMISSION_INTEGRATIONS" -value "ILogger"
+
+if ($IsWindows) {
```
Comment (zacharycmontoya) [inline tracer/samples/AzureFunctionsWithAgentLessLogging/FunctionWithLogging/run.ps1:68]: I'm on Powershell 5 (i.e. Windows / .NET Framework) and the variable `$IsWindows` isn't present. For this environment, we can use the following to run this branch:

```suggestion
if ($IsWindows -or ([System.Environment]::OSVersion.Platform -eq "Win32NT")) {
```

File: tracer/samples/AzureFunctionsWithAgentLessLogging/FunctionWithLogging/run.ps1 [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    Write-Host "Using Windows configuration"
+    $AppDirectory = Get-ApplicationDirectory
```
Comment (zacharycmontoya) [inline tracer/samples/AzureFunctionsWithAgentLessLogging/FunctionWithLogging/run.ps1:70]: I don't see `Get-ApplicationDirectory` defined anywhere. What if we change it to this?

```suggestion
    if ($null -ne $AppDirectory)
    {
        $AppDirectory = "$pwd"
    }
```

**Pull request merged**